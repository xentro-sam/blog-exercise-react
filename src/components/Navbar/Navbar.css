.navbar {
    background-color: #000000;
    border: 0;
    color: white;
    display: flex;
    flex-direction: column;
}

.navbar-links {
    display: flex;
    justify-content: space-around;
    align-self: flex-end;
}

.navbar-title > h2 {
    margin-bottom: 0;
}

.navbar-subtitle > h4 {
    margin: 0;
}

.navbar-links-items {
    padding-left: 120px;
}
