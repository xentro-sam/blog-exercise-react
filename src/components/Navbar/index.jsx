import * as React from 'react';
import './Navbar.css';

export default function Navbar() {
  return (
    <div className="navbar">
      <div className="navbar-title page-padding">
        <h2>The Artifact</h2>
      </div>
      <div className="navbar-subtitle page-padding">
        <h4><em>Culture & Art Blog</em></h4>
      </div>
      <div className="navbar-links page-padding">
        <div className="navbar-links-items">Blog</div>
        <div className="navbar-links-items">About</div>
        <div className="navbar-links-items">Contact</div>
      </div>
    </div>
  );
}
