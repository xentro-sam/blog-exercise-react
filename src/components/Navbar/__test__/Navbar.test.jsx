import {render} from '@testing-library/react';
import React from 'react';
import Navbar from '..';

describe('Navbar', () => {
  it('should render correctly', () => {
    const {asFragment} = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render the title', () => {
    const {queryByText} = render(<Navbar />);
    expect(queryByText('The Artifact')).toBeTruthy();
  });
  it('should render the subtitle', () => {
    const {queryByText} = render(<Navbar />);
    expect(queryByText('Culture & Art Blog')).toBeTruthy();
  });
  it('should render the blog link', () => {
    const {queryByText} = render(<Navbar />);
    expect(queryByText('Blog')).toBeTruthy();
  });
  it('should render the about link', () => {
    const {queryByText} = render(<Navbar />);
    expect(queryByText('About')).toBeTruthy();
  });
  it('should render the contact link', () => {
    const {queryByText} = render(<Navbar />);
    expect(queryByText('Contact')).toBeTruthy();
  });
});
