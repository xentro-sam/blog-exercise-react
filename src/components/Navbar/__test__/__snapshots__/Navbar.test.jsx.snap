// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="navbar-title page-padding"
    >
      <h2>
        The Artifact
      </h2>
    </div>
    <div
      class="navbar-subtitle page-padding"
    >
      <h4>
        <em>
          Culture & Art Blog
        </em>
      </h4>
    </div>
    <div
      class="navbar-links page-padding"
    >
      <div
        class="navbar-links-items"
      >
        Blog
      </div>
      <div
        class="navbar-links-items"
      >
        About
      </div>
      <div
        class="navbar-links-items"
      >
        Contact
      </div>
    </div>
  </div>
</DocumentFragment>
`;
